blog/02-15-2023-build-feature-rich-persistent-dapp-on-celo-using-wagmi/build-feature-rich-persistent-dapp-on-celo-using-wagmi.md

```diff
@@ -0,0 +1,747 @@
+
+Lastly, we get the already configured `celoAlfajores` network information from the `'wagmi/chains'` module. Now, the work is a lot easier. 
+
+```ts
+import { getDefaultClient } from 'connectkit'
+import { createClient } from 'wagmi'
+import { celoAlfajores } from 'wagmi/chains'
+
+export const client = createClient(
+  getDefaultClient({
+    autoConnect: true,
+    appName: 'DRythm Music-sharing dapp',
+    chains: [celoAlfajores]
+  })
+)
+```
+
+- `_app.ts`
+
+Import the client from the previous step in 'wagmi.ts'.
+Import WagmiConfig provider from 'wagmi'. We'll need this to wrap and initialize our dapp so wagmi can effectively manage it for us. It will give us access to the various wagmi hooks in the child components.
+
+Get the 'ConnectKitProvider' component from 'connectkit' module. From the name, it manages the connection and connected interface for us.
+
+>Note: With this version of connectkit, for any of its components to work anywhere in your program, it must be wrapped within a ConnectkitProvider as depicted below.
+
+```ts
+import '../styles/globals.css'
+import React from 'react';
+import NextHead from 'next/head'
+import type { AppProps } from 'next/app';
+import { client } from '../wagmi'
+import { WagmiConfig } from 'wagmi';
+import { ConnectKitProvider } from 'connectkit';
+
+export default function App({ Component, pageProps }: AppProps) {
+  const [mounted, setMounted] = React.useState(false);
+
+  React.useEffect(() => setMounted(true), []);
+
+  return (
+    <WagmiConfig client={client}>
+      <ConnectKitProvider>
+        <NextHead>
+          <title>SwapLab</title>
+        </NextHead>
+          {
+            mounted &&
+              <Component 
+                {...pageProps} 
+              />
+        }
+      </ConnectKitProvider>
+    </WagmiConfig>
+  );
+}
+```
+
+- `Header/index.tsx`
+
+In the Header component, we import and render the 'ConnectKitButton' component from 'connectkit' module. The component manages user authentication, disconnection, and account and balances display for us with a very nice user interface. It saves us a lot of stress and we only need to write less code. From the `useAccount` hook, we can access properties such as the user's connected account, ENS name, provider object, and many more.
```

Comment:
In the Header component, we import and render the 'ConnectKitButton' component from 'connectkit' module. The component manages user authentication, disconnection, and account and balances display for us with a very nice user interface. It saves us a lot of stress, and we only need to write less code. From the `useAccount` hook, we can access properties such as the user's connected account, ENS name, provider object, and many more.